### PR TITLE
feat: support avax (and other chain signing) on keepkey

### PR DESCRIPTION
**Pull request by pastaghost**
**Updated PR containing work originally done by @foufrix**

Support signing EVM-compatible chains with KeepKey.

### CODE REVIEW

**Comment by pastaghost** [comment]
I closed the original PR and opened this one with some additional changes applied. @foufrix's changes all look good to me.

File: integration/src/ethereum/ethereum.ts [hunk 1]
```diff
@@ -243,6 +243,52 @@ export function ethereumTests(get: () => { wallet: core.HDWallet; info: core.HDW
...
 
+    test(
+      "ethSignTx() - AVAX",
+      async () => {
+        if (!wallet) return;
+        // only test keepkey AVAX signing
+        // eslint-disable-next-line jest/no-jasmine-globals, jest/no-disabled-tests
+
+        const txToSign = {
+          addressNList: core.bip32ToAddressNList("m/44'/60'/0'/0/0"),
```
Comment (kaladinlight) [inline integration/src/ethereum/ethereum.ts:254]: use avalanche c chain coin type
```suggestion
          addressNList: core.bip32ToAddressNList("m/44'/9000'/0'/0/0"),
```

File: integration/src/ethereum/ethereum.ts [hunk 2]
```diff
@@ -243,6 +243,52 @@ export function ethereumTests(get: () => { wallet: core.HDWallet; info: core.HDW
       TIMEOUT
     );
 
+    test(
+      "ethSignTx() - AVAX",
+      async () => {
+        if (!wallet) return;
+        // only test keepkey AVAX signing
+        // eslint-disable-next-line jest/no-jasmine-globals, jest/no-disabled-tests
```
Comment (kaladinlight) [inline integration/src/ethereum/ethereum.ts:251]: comments don't seem applicable (wallet check will perform this test on any wallets with `supportsETH === true)`). we probably want a `supportsAVAX` for granularity 
```suggestion
```

File: integration/src/wallets/trezor.ts [hunk 3]
```diff
@@ -83,6 +83,30 @@ export class MockTransport extends trezor.TrezorTransport {
       }
     );
 
+    this.memoize(
+      "ethereumSignTransaction",
+      {
+        path: [2147483692, 2147483708, 2147483648, 0, 0],
```
Comment (kaladinlight) [inline integration/src/wallets/trezor.ts:89]:
```suggestion
        path: [2147483692, 2147492648, 2147483648, 0, 0],
```

File: integration/src/wallets/trezor.ts [hunk 4]
```diff
@@ -83,6 +83,30 @@ export class MockTransport extends trezor.TrezorTransport {
       }
     );
 
+    this.memoize(
```
Comment (kaladinlight) [inline integration/src/wallets/trezor.ts:86]:
```suggestion
    // AVAX
    this.memoize(
```

File: integration/src/wallets/ledger.ts [hunk 5]
```diff
@@ -112,6 +112,16 @@ export class MockTransport extends ledger.LedgerTransport {
           '{"success":true,"payload":{"v":"25","r":"1238fd332545415f09a01470350a5a20abc784dbf875cf58f7460560e66c597f","s":"10efa4dd6fdb381c317db8f815252c2ac0d2a883bd364901dee3dec5b7d3660a"},"coin":"Eth","method":"signTransaction"}'
         )
       );
+      this.memoize(
+        "Eth",
+        "signTransaction",
+        JSON.parse(
+          '["m/44\'/60\'/0\'/0/0","f872018501dcd6500082562294dafea492d9c6733ae3d56b7ed1adb60692c98bc5872c68af0bb14000b844a9059cbb0000000000000000000000001d8ce9022f6284c3a5c317f8f34620107214e54500000000000000000000000000000000000000000000000000000002540be40082a86a8080"]'
```
Comment (kaladinlight) [inline integration/src/wallets/ledger.ts:119]:
```suggestion
          '["m/44\'/9000\'/0\'/0/0","f872018501dcd6500082562294dafea492d9c6733ae3d56b7ed1adb60692c98bc5872c68af0bb14000b844a9059cbb0000000000000000000000001d8ce9022f6284c3a5c317f8f34620107214e54500000000000000000000000000000000000000000000000000000002540be40082a86a8080"]'
```

File: integration/src/wallets/ledger.ts [hunk 6]
```diff
@@ -112,6 +112,16 @@ export class MockTransport extends ledger.LedgerTransport {
           '{"success":true,"payload":{"v":"25","r":"1238fd332545415f09a01470350a5a20abc784dbf875cf58f7460560e66c597f","s":"10efa4dd6fdb381c317db8f815252c2ac0d2a883bd364901dee3dec5b7d3660a"},"coin":"Eth","method":"signTransaction"}'
         )
       );
+      this.memoize(
```
Comment (kaladinlight) [inline integration/src/wallets/ledger.ts:115]:
```suggestion
      // AVAX
      this.memoize(
```

File: integration/src/ethereum/ethereum.ts [hunk 7]
```diff
@@ -243,6 +243,52 @@ export function ethereumTests(get: () => { wallet: core.HDWallet; info: core.HDW
...
+
+        const txToSign = {
+          addressNList: core.bip32ToAddressNList("m/44'/60'/0'/0/0"),
+          nonce: "0x01",
+          gasPrice: "0x1dcd65000",
+          gasLimit: "0x5622",
+          value: "0x2c68af0bb14000",
```
Comment (kaladinlight) [inline integration/src/ethereum/ethereum.ts:258]: Kind of a weird test transaction with eth value send along with transfer contract data. I prefer more realistic test data, but tests pass with matching mocks as it stands so won't block over it, just making note.

**Comment by foufrix** [comment]
does the comments need some change on my end @pastaghost ?

File: integration/src/ethereum/ethereum.ts (hunk 1 above)
Comment (pastaghost) [reply integration/src/ethereum/ethereum.ts:254]: 👍

File: integration/src/ethereum/ethereum.ts (hunk 2 above)
Comment (pastaghost) [reply integration/src/ethereum/ethereum.ts:251]: This breaks the current convention used in HDWallet where `_supportsBTC` is used to indicate `_supportsUTXO`. This isn't a good convention, so this should be addressed in a larger refactor later.

File: integration/src/wallets/trezor.ts (hunk 3 above)
Comment (pastaghost) [reply integration/src/wallets/trezor.ts:89]: 👍

File: integration/src/wallets/trezor.ts (hunk 4 above)
Comment (pastaghost) [reply integration/src/wallets/trezor.ts:86]: 👍

File: integration/src/wallets/ledger.ts (hunk 5 above)
Comment (pastaghost) [reply integration/src/wallets/ledger.ts:119]: 👍

File: integration/src/wallets/ledger.ts (hunk 6 above)
Comment (pastaghost) [reply integration/src/wallets/ledger.ts:115]: 👍

File: integration/src/ethereum/ethereum.ts (hunk 7 above)
Comment (pastaghost) [reply integration/src/ethereum/ethereum.ts:258]: 👍

**Comment by pastaghost** [comment]
> do the comments need some change on my end @pastaghost?

@foufrix Nope, I made the requested changes, so we should be good to go upon approval.

**Review by kaladinlight — DISMISSED** [review]

**Review by 0xdef1cafe — DISMISSED** [review]

**Review by 0xdef1cafe — APPROVED** [review]

**Review by kaladinlight — APPROVED** [review]

**Pull request merged**